includes/api/class-wc-admin-rest-product-variations-controller.php

```diff
@@ -22,4 +22,70 @@ class WC_Admin_REST_Product_Variations_Controller extends WC_REST_Product_Variat
 	 * @var string
 	 */
 	protected $namespace = 'wc/v4';
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params           = parent::get_collection_params();
+		$params['search'] = array(
+			'description'       => __( 'Search by similar product name or sku.', 'woocommerce-admin' ),
+			'type'              => 'string',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+		return $params;
+	}
+
+	/**
+	 * Add product name and sku filtering to the WC API.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		$args = parent::prepare_objects_query( $request );
+
+		if ( ! empty( $request['search'] ) ) {
+			$args['search'] = $request['search'];
+			$args['s']      = false;
```

Comment:
Nitpick: I think it would be better to `unset()` here. _(`false` still passes an `isset()` check that other code might be using)_